File: backend/src/main/java/lan/dk/podcastserver/manager/worker/noop/NoOpExtractor.java
package lan.dk.podcastserver.manager.worker.noop;

import lan.dk.podcastserver.entity.Item;
import lan.dk.podcastserver.manager.downloader.DownloadingItem;
import lan.dk.podcastserver.manager.worker.Extractor;
import lombok.extern.slf4j.Slf4j;

import static io.vavr.API.List;

/**
 * Created by kevin on 03/12/2017
 */
@Slf4j
public class NoOpExtractor implements Extractor {

    @Override
    public DownloadingItem extract(Item item) {
        return DownloadingItem.builder()
                .item(item)
                .urls(List(item.getUrl()))
                .build();
    }

    @Override
    public Integer compatibility(String url) {
        return Integer.MAX_VALUE;
    }
}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/manager/worker/noop/NoOpExtractor.kt
package com.github.davinkevin.podcastserver.manager.worker.noop

import com.github.davinkevin.podcastserver.manager.downloader.DownloadingItem
import com.github.davinkevin.podcastserver.manager.worker.Extractor
import com.github.davinkevin.podcastserver.entity.Item

/**
 * Created by kevin on 03/12/2017
 */
class NoOpExtractor : Extractor {

    override fun extract(item: Item) =
            DownloadingItem(item, listOf(item.url!!), null, null)

    override fun compatibility(url: String?) = Integer.MAX_VALUE
}
